pkg/util/tracing/tracer.go

```diff
@@ -90,6 +90,18 @@ var ZipkinCollector = settings.RegisterStringSetting(
 	envutil.EnvOrDefaultString("COCKROACH_TEST_ZIPKIN_COLLECTOR", ""),
 ).WithPublic()
 
+var dataDogAgentAddr = settings.RegisterStringSetting(
+	"trace.datadog.agent",
+	"if set, traces will be sent to this DataDog agent; use <host>:<port> or \"default\" for localhost:8126",
```

Comment:
We only support one shadow manager, so every cluster setting will have to contain a caveat that it mutually conflicts with any of the other ones. Any appetite for fixing that limitation and unifying net/trace, lightstep, zipkin, dd, etc?

Also, can this be tested? See https://github.com/cockroachdb/cockroach/blob/2b71afb64fc7f8436085f1e6bf3b2b9bdc4f5364/pkg/util/tracing/tracer_test.go#L330